android/app/src/main/java/org/fossasia/openevent/api/APIClient.java

```diff
@@ -30,47 +39,109 @@
 
     private static final int READ_TIMEOUT_MILLIS = 50 * 1000; // 20s
 
+    private static final String CACHE_CONTROL = "Cache-Control";
+
     private static OpenEventAPI openEventAPI;
     private static FacebookGraphAPI facebookGraphAPI;
 
-    private static Retrofit.Builder retrofitBuilder;
+    private static OkHttpClient.Builder okHttpClientBuilder;
 
     static {
-        OkHttpClient.Builder okHttpClientBuilder = new OkHttpClient().newBuilder()
+        okHttpClientBuilder = new OkHttpClient().newBuilder()
                 .connectTimeout(CONNECT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS)
                 .readTimeout(READ_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
 
         if (BuildConfig.DEBUG)
             okHttpClientBuilder.addNetworkInterceptor(new StethoInterceptor());
-
-        OkHttpClient okHttpClient = okHttpClientBuilder.addInterceptor(new HttpLoggingInterceptor()
-                .setLevel(HttpLoggingInterceptor.Level.BASIC))
-                .build();
-
-        retrofitBuilder = new Retrofit.Builder()
-                .addConverterFactory(JacksonConverterFactory.create(new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)))
-                .client(okHttpClient);
     }
 
     public static OpenEventAPI getOpenEventAPI() {
-        if (openEventAPI == null)
+        if (openEventAPI == null) {
+            OkHttpClient okHttpClient = okHttpClientBuilder.addInterceptor(new HttpLoggingInterceptor()
+                    .setLevel(HttpLoggingInterceptor.Level.BASIC))
+                    .build();
+
+            Retrofit.Builder retrofitBuilder = new Retrofit.Builder()
+                    .addConverterFactory(JacksonConverterFactory.create(new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)))
+                    .client(okHttpClient);
+
             openEventAPI = retrofitBuilder
                     .baseUrl(Urls.BASE_URL)
                     .build()
                     .create(OpenEventAPI.class);
+        }
 
         return openEventAPI;
     }
 
     public static FacebookGraphAPI getFacebookGraphAPI() {
-        if (facebookGraphAPI == null)
+        if (facebookGraphAPI == null) {
+            OkHttpClient okHttpClient = okHttpClientBuilder.addInterceptor(new HttpLoggingInterceptor()
+                    .setLevel(HttpLoggingInterceptor.Level.BASIC))
+                    .addInterceptor(provideOfflineCacheInterceptor())
+                    .addNetworkInterceptor(provideCacheInterceptor())
+                    .cache(provideCache())
+                    .build();
+
+            Retrofit.Builder retrofitBuilder = new Retrofit.Builder()
+                    .addConverterFactory(JacksonConverterFactory.create(new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)))
+                    .client(okHttpClient);
```

Comment:
the okHttpClient paramter is not same. I have provided the cache and added the interceptor for the the facebook okhttpclient.